src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Minimal.cs

```diff
@@ -166,7 +165,24 @@ private void MinimallyQualify(INamedTypeSymbol symbol)
                 return SpecializedCollections.EmptyDictionary<INamespaceOrTypeSymbol, IAliasSymbol>();
             }
 
-            var token = semanticModelOpt.SyntaxTree.GetRoot().FindToken(positionOpt);
+            // Walk up the ancestors from the current position. If this is a speculative
+            // model, walk up the corresponding ancestors in the parent model.
+            SemanticModel semanticModel;
+            int position;
+            if (semanticModelOpt.IsSpeculativeSemanticModel)
+            {
+                var methodRoot = ((MethodBodySemanticModel)semanticModelOpt).Root;
```

Comment:
It may not be safe after all. However, it seems the cast can be changed to `CSharpSemanticModel`.